### ISSUE TITLE
Attribute contribution commits to the contributor

### ISSUE THREAD

**Issue by storrisi** [clearlydefined/service#546]
See: https://github.com/clearlydefined/curated-data/pull/1864/commits

Even if the PR has been opened by a contributor, commits result to be attributed to clearlydefinedbot. 

We want to attribute also the commits to the owner of the PR.

**Comment by ignacionr** [comment]
Just made a test and the commit ended up correctly attributed.
The test commit was https://github.com/clearlydefined/curated-data-dev/pull/915/commits/2b52d005cb06b11ff3150047c1db5bd23da40246 

The code that takes care of this comes from https://github.com/clearlydefined/service/commit/dac01cab6ff70dba20f0be5ef0cd102f1a7bf96a#diff-d081ea700bee378a703ee270043c1a1fR173 

If somebody can help reproduce it, much appreciated.

**Comment by storrisi** [comment]
@ignacionr is that a recent change? See https://github.com/clearlydefined/curated-data-dev/pull/912/commits

**Comment by ignacionr** [comment]
@storrisi We need to understand that PRs and commits have completely different attribution mechanics.

- For our system to create a PR attributed to the user, we only need to know the user and have their permission.
- For us to attribute a commit, we need two pieces of information that are totally independent on the GitHub user: a name, and an email. Dan decided to attribute exclusively if we had both pieces of data. Those pieces of data are accessible through the API with the user token, and such access works in normal cases (as I have recently tested out). But the API could not have been accessible at the moment, the token could have been restricted, or more simply, Gene may not be listing his email address.

The change is not recent, but it requires the GitHub user to be listing an email address (as accessible by the token); it may be the case that Gene doesn't have one listed, or also maybe at the precise moment of the creation of the commit, the GitHub API didn't respond for some reason.
As shown, I have created a curation successfully and it shows up ok.

**Comment by storrisi** [comment]
@ignacionr thank for clarifying it. 
I think that if we want to be sure that also the commits would be attributed to the contributor, then we should ask for the email of the user just before the contribution, if that is not provided or is not available in his public github profile.

What do you think?

**Comment by ignacionr** [comment]
Yes, I think we need to either ask for the email or let it be known that without an email that's visible on profile, we can't do the attribution part.

**Comment by storrisi** [comment]
As per meeting discussion, we will add a message in UI to let the user know the commits/contribution will not be attributed to them, but let them continue to make the contribution without their information provided.

Moving this to the Website repo.
